fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/AbstractPersister.java

```diff
@@ -116,18 +120,45 @@ protected FedoraOCFLMapping getMapping(final String resourceId) throws Persisten
     protected String resolveRootObjectId(final CreateResourceOperation operation,
                                        final OCFLPersistentStorageSession session) {
 
-        final var parentId = operation.getParentId();
         final var resourceId = operation.getResourceId();
-        //final ResourceHeaders headers = session.getHeaders(parentId, null);
-        final boolean parentIsAg  = false; // TODO uncomment when headers.isAchivalGroup() is available
-        if (parentIsAg) {
-            return parentId;
+        //is resource or any parent an archival group?
+        final var startingResourceId = operation.getType().equals(CREATE) ? operation.getParentId() : resourceId;
+        final var archivalGroupId = findArchivalGroupInAncestry(startingResourceId, session);
+
+        if (archivalGroupId != null) {
+            return archivalGroupId;
         } else if (resourceId.endsWith("/" + FCR_METADATA) || resourceId.endsWith("/" + FCR_ACL)) {
             return resourceId.substring(0, resourceId.lastIndexOf("/"));
         } else {
             return resourceId;
         }
+}
+
+    protected String findArchivalGroupInAncestry(final String resourceId, final OCFLPersistentStorageSession session) {
+            if (resourceId == null || resourceId.endsWith(FEDORA_ID_PREFIX)) {
+                return null;
+            }
+
+            //strip off trailing slash if exists
+            String cleanedResourceId = resourceId;
+            if (resourceId.endsWith("/")) {
+                cleanedResourceId = resourceId.substring(0, resourceId.length() - 1);
+            }
+
+            try {
+                final var headers = session.getHeaders(cleanedResourceId, null);
+                if (headers.isArchivalGroup()) {
+                    return cleanedResourceId;
+                }
+            } catch (final PersistentItemNotFoundException ex) {
+                //do nothing
```

Comment:
What is the scenario in which this would be thrown? It might be good to document why its okay for the resource to now be found.